### PR TITLE
Misc. fixes and refactoring

### PR DESCRIPTION
**Pull request by safl**
While experimenting with code-generation (https://github.com/OpenMPDK/xNVMe/pull/348) of the xNVMe C API and FFIs for Python and Rust, then several challenges appeared. These are addressed by refactoring. Misc. fixes are also provided which addresses build-warnings. more will come, however, no reason to hold these back.

### CODE REVIEW

**Review by karlowich — DISMISSED** [review]
This looks good!
I agree that this adds more clarity to the code

File: lib/xnvme_spec.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ xnvme_spec_log_health_fpr(FILE *stream, const struct xnvme_spec_log_health_entry
 
 	wrtn += fprintf(stream, "\n");
 	wrtn += fprintf(stream, "  crit_warn: %" PRIu8 "\n", log->crit_warn);
-	wrtn += fprintf(stream, "  comp_temp: %" PRIu16 "\n", log->comp_temp - 273u);
+	wrtn += fprintf(stream, "  comp_temp: %" PRIu16 "\n", (uint16_t)(log->comp_temp - 273u));
```
Comment (birkelund) [inline lib/xnvme_spec.c:46]: By subtracting 273 you are converting from K to C. The result may very well be negative. Just let the compiler promote `comp_temp` to `int`.

File: lib/xnvme_spec.c [hunk 2]
```diff
@@ -67,7 +67,7 @@ xnvme_spec_log_health_fpr(FILE *stream, const struct xnvme_spec_log_health_entry
 	wrtn += fprintf(stream, "  crit_comp_temp_time: %" PRIu32 "\n", log->crit_comp_temp_time);
 	for (int i = 0; i < 8; ++i) {
 		wrtn += fprintf(stream, "  temp_sens%u: %" PRIu16 "\n", i + 1,
-				log->temp_sens[i] ? log->temp_sens[i] - 273u : 0);
+				(uint16_t)(log->temp_sens[i] ? log->temp_sens[i] - 273u : 0));
```
Comment (birkelund) [inline lib/xnvme_spec.c:70]: Same here.

Comment (safl) [reply lib/xnvme_spec.c]: Right, thanks! Fixed.

File: lib/xnvme_spec.c (hunk 1 above)
Comment (safl) [reply lib/xnvme_spec.c]: Right, thanks! Fixed.

**Pull request merged**